.github/workflows/lint.yml

```diff
@@ -0,0 +1,52 @@
+name: Linter
+
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: Lint Codebase
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        language: [python, javascript]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      # Setup and lint Python code
+      - name: Setup Python
+        if: matrix.language == 'python'
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install Python dependencies
+        if: matrix.language == 'python'
+        run: |
+          python -m pip install --upgrade pip
+          pip install ruff
+
+      - name: Run Ruff
+        if: matrix.language == 'python'
```

Comment:
I see that the benefit is that it would run the steps in parallel.. I found this:

```
In GitHub Actions, the matrix syntax allows you to define a set of jobs that run multiple configurations in parallel, which is useful for testing across different environments, dependencies, or parameters. The matrix syntax helps you automate running the same workflow against different combinations of variables without having to manually define each combination.

Example Use Cases for Matrix Builds:
- Running your tests on multiple versions of Node.js, Python, or other runtimes.
- Running your application on multiple operating systems (e.g., Ubuntu, Windows, macOS).
- Testing with different versions of dependencies or databases.
```

I am wondering if `matrix` is the right tool for this particular scenario, however, a point can be made to keep the matrix for the performance boost gained by them running in parallel